src/Aspire.Hosting/Redis/RedisCommanderConfigWriterHook.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using System.Text;
+using Aspire.Hosting.Lifecycle;
+
+namespace Aspire.Hosting.Redis;
+
+public class RedisCommanderConfigWriterHook : IDistributedApplicationLifecycleHook
+{
+    public Task AfterEndpointsAllocatedAsync(DistributedApplicationModel appModel, CancellationToken cancellationToken)
+    {
+        var commanderResource = appModel.Resources.OfType<RedisCommanderResource>().Single();
```

Comment:
This should be more defensive (SingleOrDefault).

Reply:
I believe `Single(...)` is correct. You can't have this lifecycle hook added unless the resource was also added. If it is, then its exceptional :)

Reply:
If another one was added that removes this resource from the list (for some obscure reason maybe), this should noop.

Reply:
Added no-ops.